ArcGISRuntimeSDKQt_CppSamples/Routing/FindClosestFacilityToMultipleIncidentsService/FindClosestFacilityToMultipleIncidentsService.cpp

```diff
@@ -0,0 +1,241 @@
+  m_incidentsFeatureTable->load();
+
+}
+
+void FindClosestFacilityToMultipleIncidentsService::setupRouting()
+{
+  // connect to createDefaultParametersCompleted signal of the closest facility task
+  connect(m_task, &ClosestFacilityTask::createDefaultParametersCompleted, this, [this]
+          (QUuid, Esri::ArcGISRuntime::ClosestFacilityParameters defaultParameters)
+  {
+    // create query parameters to query features from feature table
+    QueryParameters params;
+    params.setWhereClause("1=1");
+    m_facilityParams = defaultParameters;
+    // set closest facilities parameters facilities
+    m_facilityParams.setFacilitiesWithFeatureTable(m_facilitiesFeatureTable, params);
+    // set closest facilitites parameters incidents
+    m_facilityParams.setIncidentsWithFeatureTable(m_incidentsFeatureTable, params);
+
+    // disable busy indicator
+    m_busy = false;
+    // enable solve button
+    m_solveButton = true;
+    emit busyChanged();
+    emit solveButtonChanged();
+  });
+
+  // connect to solveClosestFacilityCompleted signal of the closest facility task
+  connect(m_task, &ClosestFacilityTask::solveClosestFacilityCompleted, this, [this]
+          (QUuid, ClosestFacilityResult closestFacilityResult)
+  {
+    if(closestFacilityResult.isEmpty())
+    {
+      qDebug() << "Empty result";
+      return;
+    }
+
+    // loop through all incidents
+    for(int incidentIndex = 0; incidentIndex < m_incidentsFeatureTable->numberOfFeatures(); incidentIndex++)
+    {
+      // get the index of the facility closest to the incident
+      int closestFacilityIndex = closestFacilityResult.rankedFacilityIndexes(incidentIndex).first();
+      // get the route between the incident and the facility
+      ClosestFacilityRoute route = closestFacilityResult.route(closestFacilityIndex,incidentIndex);
+      // create a graphic from the routes geometry using the route symbol
+      Graphic* m_routeGraphic = new Graphic(route.routeGeometry(), m_routeSymbol, this);
+      // append the graphic to the graphics overlay
+      m_resultsOverlay->graphics()->append(m_routeGraphic);
+    }
+
+    // append the graphics overlay to the mapview
+    m_mapView->graphicsOverlays()->append(m_resultsOverlay);
+    // enable the reset button
+    m_resetButton = true;
+    emit resetButtonChanged();
+    // disable the busy indicator
+    m_busy = false;
+    emit busyChanged();
+  });
+
+  // create default parameters for the closest facility task
+  m_task->createDefaultParameters();
+}
+
+void FindClosestFacilityToMultipleIncidentsService::setViewpointGeometry(const Error& e)
+{
+  if (!e.isEmpty())
+  {
+    qDebug() << e.message();
+    return;
+  }
+
+  // return if either or both layers are not loaded
+  if ((m_facilitiesFeatureTable->loadStatus() != LoadStatus::Loaded) || (m_incidentsFeatureTable->loadStatus() != LoadStatus::Loaded))
+    return;
+
+  // return if set viewpoint task watcher has been created
+  if (setViewpointTaskWatcher)
```

Comment:
See other comment, this should not be a pointer type, in which case this would change to `if (setViewpointTaskWatcher.isValid())`